docs/config/lua/window/get_appearance.md

```diff
@@ -45,4 +45,40 @@ end)
 return {
 }
 ```
+### Wayland GNOME Workaround
+
+If you happen to be running a Wayland session through GNOME desktop,
+a workaround exists in which you can query the `gsettings` command
+for the theme name. While GNOME does not specifically support
+appearance, it is common for themes to provide a dark variant (and
+this is the case for the default "Adwaita" theme), which are typically
+named with "-dark" appended to the theme name. With this heuristic,
+we can determine if GNOME is in dark mode or not, and abuse the
+[update-right-status](../window-events/update-right-status.md)
+event to poll for changes. When coupled with the [Night
+Theme](https://nightthemeswitcher.romainvigier.fr/) extension or
+similar, you can enjoy a Wezterm that changes theme based on day night
+cycle with the rest of your system.
+
+```lua
+function·compute_scheme_gnome_wayland()¬
+··local·success,·stdout·=·wezterm.run_child_process(¬
+····{"gsettings",·"get",·"org.gnome.desktop.interface",·"gtk-theme"}¬
```

Comment:
I think it would be helpful to factor out the bit that determines the appearance.

For example:

```lua
function query_appearance_gnome()
   local success, stdout = wezterm.run_child_process({"gsettings", ....})
   -- lowercase and remove whitespace
   stdout = stdout.lower():gsub("%s+", "")
   local mapping = {
      highcontrast = "LightHighContrast",
      highcontrastinverse = "DarkHighContrast",
      adwaita = "Light",
      ["adwaita-dark"] = "Dark"
   }
   if stdout == "highcontrast" then
     return "LighHighContrast"
   end
   local appearance = mapping[stdout]
   if appearance then
      return appearance
   end
   if stdout:find("dark") then
      return "Dark"
   end
   return "Light"
end
```

See also: https://github.com/wez/wezterm/blob/main/window/src/os/x11/connection.rs#L170-L175

Then the example can reuse the `scheme_for_appearance` function from the earlier example on this page, and use the `update-right-status` hook using:

```lua
  local appearance = query_appearance_gnome()
  local scheme = scheme_for_appearance(appearance)
```

Reply:
I tested this out locally and after a little debugging this works well even with the High Contrast switch in GNOME's Accessibility menu. :100:  Refactored in latest commit.